doc/10-Channels.md

```diff
@@ -1,88 +1,298 @@
+  ```json
+  {
+    "error": "unknown method: 'Foo'",
+    "id": 1000
+  }
+  ```
+
+### Methods
+
+The following methods must be implemented by a channel.
+
+#### GetInfo
+
+The parameterless `GetInfo` method returns information about the channel.
+
+Its `result` is expected to be a JSON object with the `json` fields defined in the
+[`Info` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#Info).
+The `config_attrs` field must be an array of JSON objects according to the
+[`ConfigOption` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#ConfigOption).
+Those attributes define configuration options for the channel to be set via the `SetConfig` method.
+Furthermore, they are used for channel configuration in Icinga Notifications Web.
+
+##### Example GetInfo Request
+
 ```json
 {
-  "method": "Add",
-  "params": {
-    "num1": 5,
-    "num2": 3
+  "method": "GetInfo",
+  "id" : 1
+}
+```
+
+##### Example GetInfo Response
+
+```json
+{
+  "result": {
+    "name": "Minified Webhook",
+    "version": "0.0.0-gf369a11-dirty",
+    "author": "Icinga GmbH",
+    "config_attrs": [
+      {
+        "name": "url_template",
+        "type": "string",
+        "label": {
+          "de_DE": "URL-Template",
+          "en_US": "URL Template"
+        },
+        "help": {
+          "de_DE": "URL, optional als Go-Template über das zu verarbeitende plugin.NotificationRequest.",
+          "en_US": "URL, optionally as a Go template over the current plugin.NotificationRequest."
+        },
+        "required": true,
+        "min": null,
+        "max": null
+      },
+      {
+        "name": "response_status_codes",
+        "type": "string",
+        "label": {
+          "de_DE": "Antwort-Status-Codes",
+          "en_US": "Response Status Codes"
+        },
+        "help": {
+          "de_DE": "Kommaseparierte Liste erwarteter Status-Code der HTTP-Antwort, z.B.: 200,201,202,208,418",
+          "en_US": "Comma separated list of expected HTTP response status code, e.g., 200,201,202,208,418"
+        },
+        "default": "200",
+        "min": null,
+        "max": null
+      }
+    ]
   },
-  "id": 2020
+  "id": 1
 }
 ```
 
+#### SetConfig
+
+The `SetConfig` method configures the channel and
+should be called at least once before sending the first notifications to initialize the channel plugin.
+
+The passed JSON object in the request's `param` field reflects the objects from `GetInfo`'s `config_attrs`.
+Each object within the `config_attrs` array can be configured by using its `name` attribute as the key together with the
+desired configuration value, being of the specified type in the `type` field.
+
+To illustrate, the URL template from the above output is configurable with the following JSON object passed in `params`:
+
 ```json
 {
-  "method": "Foo",
+  "url_template": "http://localhost:8000/update/{{.Incident.Id}}"
+}
+```
+
+The response might carry an error if the configuration is invalid.
```

Comment:
What does "might" mean here? Optional? Recommended?

Might I suggest using the key words from RFC2119?

Reply:
`s/might/MAY/g` :^)

I rewrote some sections to remove inaccuracies, use some RFC 2119 language (just w/o the caps), and to address a channel developer.